hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -1935,6 +1935,21 @@ public DatanodeInfo[] getSlowDatanodeReport() throws IOException {
     return rpcServer.getSlowDatanodeReport(true, 0);
   }
 
+  @Override
+  public String getEnclosingRoot(String src) throws IOException {
+    Path mountPath = new Path("/");
```

Comment:
The default nameservice may not be turned on at the router and thus we can not assume the default mountPath to be /. There is a config to turn it on. dfs.federation.router.default.nameservice.enable. If the default nameservice is not turned on and src does not belong to any mount point, maybe we should throw an IOException here?

Reply:
Thanks for the info. I added some checking here. Still need to do something similar for viewfs since it may not have a linkfallback configured